src/azure-cli/azure/cli/command_modules/acr/helm.py

```diff
@@ -175,6 +182,81 @@ def acr_helm_repo_add(cmd,
     p.wait()
 
 
+def acr_helm_install_cli(client_version='2.16.3', install_location=None, yes=False):
+    """Install Helm command-line tool."""
+
+    if client_version >= '3':
+        raise CLIError('Helm v3 is not supported yet.')
+
+    if not install_location or install_location.isspace():
+        raise CLIError('Invalid install location.')
+
+    install_dir, cli = os.path.dirname(install_location), os.path.basename(install_location)
+    if not cli:
+        raise CLIError("{} is a directory. Please specify a complete file path.".format(install_location))
+
+    # Ensure installation directory exists
+    if not os.path.exists(install_dir):
+        os.makedirs(install_dir)
+
+    client_version = "v%s" % client_version
+    source_url = 'https://get.helm.sh/{}'
+    package, folder = _get_helm_package_name(client_version)
+    download_path = ''
+
+    if not package:
+        raise CLIError('No prebuilt binary for current system.')
+
+    try:
+        import tempfile
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            download_path = os.path.join(tmp_dir, package)
+            _urlretrieve(source_url.format(package), download_path)
+            _unzip(download_path, tmp_dir)
+
+            sub_dir = os.path.join(tmp_dir, folder)
+            # Ask user to check license
+            if not yes:
+                with open(os.path.join(sub_dir, 'LICENSE')) as f:
+                    text = f.read()
+                    logger.warning(text)
+                user_confirmation('Before proceeding with the installation, '
+                                  'please confirm that you have read and agreed the above license.')
+
+            # Move files from temporary location to specified location
+            import shutil
+            import stat
+            for f in os.scandir(sub_dir):
+                # Rename helm to specified name
+                target_path = install_location if os.path.splitext(f.name)[0] == 'helm' \
+                    else os.path.join(install_dir, f.name)
+                logger.debug('Moving %s to %s', f.path, target_path)
+                shutil.move(f.path, target_path)
+
+                if os.path.splitext(f.name)[0] in ('helm', 'tiller'):
+                    os.chmod(target_path, os.stat(target_path).st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
+    except IOError as e:
+        raise CLIError('Error while installing {} to {}: {}'.format(cli, install_dir, e))
```

Comment:
will this cause the original stack trace to lost? Otherwise I suggest we catch a particular line which might throw this exception, or just start with uncaught.

Reply:
We can log the stack trace so that it won't lose and users won't get disturbed